text/0000-monorepo.md

```diff
@@ -0,0 +1,359 @@
+
+At a high-level, we would use GitHub's [`CODEOWNERS`](https://help.github.com/en/articles/about-code-owners) functionality to assign GitHub teams to specific parts of the project. These teams would be aliases of our workstreams, most likely. This would trigger required reviews
+
+Push access to the project would be given to all members of the GitHub teams included in the project.
+
+Publish access would ideally be automated through CI, but manual access can be granted to packages that belong to a specific GitHub team / workstream.
+
+## Add-ons
+
+Within this proposal, the role of an add-on will most likely need to be re-defined. Currently, an add-on exists as a supplement to the Carbon Design System. This means that teams arrive at solutions that they need that don't exist in Carbon currently, and they have a desire to share within their team. The add-on model works well for this direct use-case, but unfortunately when scaling out Carbon to all of IBM this model has a habit of entrenching behavior.
+
+For example, imagine we had teams A, B, and C all working on platforms. Team A comes up with a `PlatformHeader` component that teams B and C would like to use.
+
+Team B can use the component as-is, which is great. However, Team C would like some current functionality tweaked to meet their use-case. While teams A and C could work together to arrive at a joint resolution, often times deadlines and scope limit this kind of collaboration.
+
+In addition, the experience of the developer can be confusing in these moments. The packages for these add-ons would like be:
+
+```
+# Team A
+carbon-addons-a
+carbon-addons-b
+carbon-addons-c
+```
+
+In the existing landscape, team B would need to install (at minimum) the following:
+
+```bash
+yarn add carbon-components \
+  carbon-components-react \ 
+  carbon-icons \
+  carbon-addons-a \
+  carbon-addons-a-react \
+  carbon-addons-b \
+  carbon-addons-b-react 
+```
+
+Ideally, the best experience for this group would be a single `carbon-addons-b` or `carbon-addons-b-react`.
+
+The other downside to this is the aspect of who owns support and who decides what is in-scope for enhancements of a component. When components are built inside add-ons, this can fall on the team whose add-on houses the component. However, if another team is requesting enhancements most likely they should be the ones to implement (when reasonable).
+
+As a result, the proposal is that **all** component development happens in the Carbon repo and add-on packages move to becoming **the defacto way to use Carbon for a Business unit, portfolio, offering, or team**. What this means is that add-ons move increasingly towards their domain, and components like `PlatformHeader` should move to the main Carbon repo.
```

Comment:
This seems like a "try it and see" situation ... there will definitely be resistance (my reaction was a bit of a roller coaster!). 

How will PR's be handled in this case? Obviously we want to keep code coming in consistent with our general standards, but depending on the PR volume, adequate review could consume quite a bit of time ... Perhaps onboard community members as contributors?